src/AutoRest.CSharp/LowLevel/Output/OperationConfidentChecker.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using AutoRest.CSharp.Common.Input;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace AutoRest.CSharp.Output.Models
+{
+    internal static class OperationConfidentChecker
+    {
+        private static ConcurrentDictionary<InputType, bool> _cache = new();
+
+        public static bool IsConfident(InputOperation operation)
+        {
+            // check parameters
+            foreach (var parameter in operation.Parameters)
+            {
+                // skip the special parameters since some particular types are still interpreted as unions (for instance content type could have union of literal types which is actually enums)
+                if (parameter.IsContentType || parameter.IsApiVersion || parameter.IsEndpoint)
+                    continue;
+
+                var isConfident = IsConfident(parameter.Type);
+                if (!isConfident)
+                    return false;
+            }
+            // check response
+            foreach (var response in operation.Responses)
```

Comment:
```suggestion
            if (operations.Responses.Any(r => !IsConfident(r.BodyType)))
            {
                return false;
            }
```

Reply:
For readability and the ability to clearly debugging this, I do not want to change it using Linq here.